api/v1alpha1/instaslice_types.go

```diff
@@ -56,105 +33,161 @@ const (
 	AllocationStatusCreated  AllocationStatus = "created"
 )
 
-// Define the struct for allocation details
-type AllocationDetails struct {
-	// profile requested by user workload
-	// +required
-	Profile string `json:"profile"`
-	// start position of a profile on a gpu
-	// +required
-	Start int32 `json:"start"`
-	// size of profile that begins from start position
-	// +required
-	Size int32 `json:"size"`
-	// podUUID represents uuid of user workload
-	// +required
-	PodUUID string `json:"podUUID"`
-	// gpuUUID represents gpu uuid of selected gpu
-	// +required
+type MigSliceProfile string
+
+const (
+	Profile1G5GB   MigSliceProfile = "1g.5gb"
+	Profile2G10GB  MigSliceProfile = "2g.10gb"
+	Profile3G20GB  MigSliceProfile = "3g.20gb"
+	Profile4G20GB  MigSliceProfile = "4g.20gb"
+	Profile7G40GB  MigSliceProfile = "7g.40gb"
+	Profile1G5GBME MigSliceProfile = "1g.5gb+me"
+	Profile1G10GB  MigSliceProfile = "1g.10gb"
+)
+
+type AllocationRequest struct {
+	// profile specifies the MIG slice profile for allocation
+	// +kubebuilder:validation:Required
+	Profile MigSliceProfile `json:"profile"`
+
+	// resources specifies resource requirements for the allocation
+	// +kubebuilder:validation:Required
+	Resources corev1.ResourceRequirements `json:"resources"`
+
+	// podRef is a reference to the Pod requesting the allocation
+	// +kubebuilder:validation:Required
+	PodRef corev1.ObjectReference `json:"podRef"`
+}
+
+type AllocationResult struct {
+	// migPlacement specifies the MIG placement details
+	// +kubebuilder:validation:Required
+	MigPlacement Placement `json:"migPlacement"`
+
+	// gpuUUID represents the UUID of the selected GPU
+	// +kubebuilder:validation:Required
 	GPUUUID string `json:"gpuUUID"`
-	// nodename represents name of the selected node
-	// +required
-	Nodename string `json:"nodename"`
-	// allocationStatus represents status of allocation
-	// +kubebuilder:validation:Enum:=deleted;deleting;ungated;creating;created
-	// +required
-	Allocationstatus AllocationStatus `json:"allocationStatus"`
-	// resourceIdentifier represents uuid used for creating configmap resource
-	// +required
-	Resourceidentifier string `json:"resourceIdentifier"`
-	// namespace represents namespace of user workload
-	// +required
-	Namespace string `json:"namespace"`
-	// podName represents name of the user workload pod
-	// +required
-	PodName string `json:"podName"`
-	// cpu represents amount of cpu requested by user workload
-	// +required
-	Cpu int64 `json:"cpu"`
-	// memory represents amount of memory requested by user workload
-	// +required
-	Memory int64 `json:"memory"`
+
+	// nodename represents the name of the selected node
+	// +kubebuilder:validation:Required
+	Nodename types.NodeName `json:"nodename"`
+
+	// allocationStatus represents the current status of the allocation
+	// +kubebuilder:validation:Enum=deleted;deleting;ungated;creating;created
+	// +kubebuilder:validation:Required
+	AllocationStatus AllocationStatus `json:"allocationStatus"`
```

Comment:
Should we create a slice of allocationstatus to record history of status change or add it to conditions?

Reply:
The transitions will be recorded in the conditions slice.